docs/cookbook/getInstantWithLocalTimeInZone.mjs

```diff
@@ -0,0 +1,92 @@
+/**
+ * Get an absolute time corresponding with a calendar date / wall-clock time in
+ * a particular time zone, the same as Temporal.TimeZone.getAbsoluteFor() or
+ * Temporal.DateTime.inTimeZone(), but with more disambiguation options.
+ *
+ * As well as the default Temporal disambiguation options 'earlier', 'later',
+ * and 'reject', there are additional options possible:
+ *
+ * - 'earlierLater': Same as what the Moment Timezone and Luxon libraries do;
+ *   equivalent to 'earlier' when turning the clock back, and 'later' when
+ *   setting the clock forward.
+ * - 'clipEarlier': Equivalent to 'earlier' when turning the clock back, and
+ *   when setting the clock forward returns the time just before the clock
+ *   changes.
+ * - 'clipLater': Equivalent to 'later' when turning the clock back, and when
+ *   setting the clock forward returns the exact time of the clock change.
+ *
+ * @param {Temporal.DateTime} dateTime - Calendar date and wall-clock time to
+ *   convert
+ * @param {Temporal.TimeZone} timeZone - Time zone in which to consider the
+ *   wall-clock time
+ * @param {string} disambiguation - Disambiguation mode, see description.
+ * @returns {Temporal.Absolute} Absolute time in timeZone at the time of the
+ *   calendar date and wall-clock time from dateTime
+ */
+function getInstantWithLocalTimeInZone(dateTime, timeZone, disambiguation = 'earlier') {
+  // Handle the built-in modes first
+  if (['earlier', 'later', 'reject'].includes(disambiguation)) {
+    return timeZone.getAbsoluteFor(dateTime, { disambiguation });
+  }
+
+  const possible = timeZone.getPossibleAbsolutesFor(dateTime);
+
+  // Return only possibility if no disambiguation needed
+  if (possible.length === 1) return possible[0];
+
+  switch (disambiguation) {
+    case 'earlierLater':
+      if (possible.length === 0) return dateTime.plus({ hours: 1 }).inTimeZone(timeZone);
```

Comment:
Should you handle time zones that potentially switch by something other than `{ hours: 1 }`?

Reply:
True, maybe adding a day would be safer.

Reply:
...or I could just let Temporal do it.